### DISCUSSION TITLE
Automate Login and fill out from web site form and extract table

### DISCUSSION THREAD

**Discussion by PabloHacker-Tools** [community/community#147648] [Programming Help]
### Body

I am encountering an issue with Selenium while automating the login process and subsequent form interactions for a web application. The script works as expected for opening the login page and locating the input fields. However, I am consistently running into the following error during login:

yaml
Code kopieren
Error during login: Message: element not interactable
(Session info: MicrosoftEdge=131.0.2903.99)

Environment Details:
Browser: Microsoft Edge (Version 131.0.2903.99)
Selenium Version: [Include your Selenium version]
Operating System: [Include your OS version]
Driver Path: C:\Users\PHacker\Documents\msedgedriver.exe

python
Code kopieren
from selenium import webdriver
from selenium.webdriver.edge.service import Service
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
from dotenv import load_dotenv
import os

# Load credentials
load_dotenv(dotenv_path='bash.env')
username = os.getenv("LOGWIN_USERNAME")
password = os.getenv("LOGWIN_PASSWORD")

# Path to EdgeDriver
driver_path = r'C:\Users\PHacker\Documents\msedgedriver.exe'
service = Service(driver_path)
driver = webdriver.Edge(service=service)

# Web application URL
url = "https://lots.logwin-logistics.com/scm/dynamic.xhtml"

try:
    # Open login page
    driver.get(url)

    # Wait for elements to load
    username_input = WebDriverWait(driver, 10).until(
        EC.presence_of_element_located((By.ID, "loginForm:username"))
    )
    password_input = driver.find_element(By.ID, "loginForm:password")
    stay_signed_in_checkbox = driver.find_element(By.CLASS_NAME, "ui-chkbox-box")
    login_button = driver.find_element(By.XPATH, "//span[text()='Login']")

    # Fill in credentials
    username_input.send_keys(username)
    password_input.send_keys(password)

    # Click "Stay signed in"
    stay_signed_in_checkbox.click()

    # Click login button
    login_button.click()

    # Wait for redirection
    WebDriverWait(driver, 10).until(EC.url_changes(url))
    print("Login successful. Current URL:", driver.current_url)

    # Perform post-login actions
    # Example: Interact with the search form
    status_dropdown = WebDriverWait(driver, 10).until(
        EC.presence_of_element_located((By.ID, "searchForm:dynamicSearchCtl:searchfields:0:dropDownStringValue"))
    )
    status_dropdown.click()

    # Select a value from the dropdown
    dropdown_option = WebDriverWait(driver, 10).until(
        EC.element_to_be_clickable((By.XPATH, "//li[@data-label='Arrived']"))
    )
    dropdown_option.click()

    # Fill in other fields
    shipment_input = driver.find_element(By.ID, "searchForm:dynamicSearchCtl:searchfields:2:textValue")
    shipment_input.send_keys("123456")

    # Submit the form
    search_button = driver.find_element(By.ID, "searchForm:searchBtn2")
    search_button.click()

    print("Form filled and submitted.")

except Exception as e:
    print("Error:", e)

finally:
    driver.quit()

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by imrkaofficial** [comment]
Updated Script example:



`from selenium import webdriver
from selenium.webdriver.edge.service import Service
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
from dotenv import load_dotenv
import os

# Load credentials
load_dotenv(dotenv_path='bash.env')
username = os.getenv("LOGWIN_USERNAME")
password = os.getenv("LOGWIN_PASSWORD")

# Path to EdgeDriver
driver_path = r'C:\Users\PHacker\Documents\msedgedriver.exe'
service = Service(driver_path)
driver = webdriver.Edge(service=service)

# Web application URL
url = "https://lots.logwin-logistics.com/scm/dynamic.xhtml"

try:
    # Open login page
    driver.get(url)

    # Wait for username input to be interactable
    username_input = WebDriverWait(driver, 10).until(
        EC.visibility_of_element_located((By.ID, "loginForm:username"))
    )
    password_input = WebDriverWait(driver, 10).until(
        EC.visibility_of_element_located((By.ID, "loginForm:password"))
    )
    stay_signed_in_checkbox = WebDriverWait(driver, 10).until(
        EC.element_to_be_clickable((By.CLASS_NAME, "ui-chkbox-box"))
    )
    login_button = WebDriverWait(driver, 10).until(
        EC.element_to_be_clickable((By.XPATH, "//span[text()='Login']"))
    )

    # Fill in credentials
    username_input.send_keys(username)
    password_input.send_keys(password)

    # Click "Stay signed in"
    stay_signed_in_checkbox.click()

    # Click login button
    login_button.click()

    # Wait for redirection
    WebDriverWait(driver, 10).until(EC.url_changes(url))
    print("Login successful. Current URL:", driver.current_url)

    # Perform post-login actions
    # Example: Interact with the search form
    status_dropdown = WebDriverWait(driver, 10).until(
        EC.visibility_of_element_located((By.ID, "searchForm:dynamicSearchCtl:searchfields:0:dropDownStringValue"))
    )
    status_dropdown.click()

    # Select a value from the dropdown
    dropdown_option = WebDriverWait(driver, 10).until(
        EC.element_to_be_clickable((By.XPATH, "//li[@data-label='Arrived']"))
    )
    dropdown_option.click()

    # Fill in other fields
    shipment_input = WebDriverWait(driver, 10).until(
        EC.visibility_of_element_located((By.ID, "searchForm:dynamicSearchCtl:searchfields:2:textValue"))
    )
    shipment_input.send_keys("123456")

    # Submit the form
    search_button = WebDriverWait(driver, 10).until(
        EC.element_to_be_clickable((By.ID, "searchForm:searchBtn2"))
    )
    search_button.click()

    print("Form filled and submitted.")

except Exception as e:
    print("Error:", e)

finally:
    driver.quit()
`
